src/util/rocksdb/properties.rs

```diff
@@ -325,6 +325,28 @@ impl SizeProperties {
         self.index_handles
             .get_approximate_distance_in_range(start, end)
     }
+
+    pub fn smallest_key(&self) -> Option<Vec<u8>> {
+        if self.index_handles.0.is_empty() {
```

Comment:
`.iter().next()` will be `None` if it is empty, so it is not necessary to do an extra check here.